### PR TITLE
Flesh out documentation

### PR DESCRIPTION
**Pull request by jeddy3**
1. I've tried to capture some of the decisions made in #46 and https://github.com/stylelint/stylelint/issues/40#issuecomment-100987199 - to keep things terse, and to save me some time, I've only captured the outcome of the decisions rather than the rationale itself. 
2. I've added a rule configuration bit to the usage part of the readme.

OK with you two?

### CODE REVIEW

File: docs/rule-guidelines.md [hunk 1]
```diff
@@ -0,0 +1,93 @@
+# Rule guidelines
+
+* Rules are standalone.
+* Rules are unopinionated:
+  * All rules are turn off by default.
```
Comment (davidtheclark) [inline docs/rule-guidelines.md]: Typo: turn -> turned

File: docs/rule-guidelines.md [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+  * All rules do *not* have default values for options.
+
+## Naming rules
+
+Rules are:
```
Comment (davidtheclark) [inline docs/rule-guidelines.md]: How about "Rule names"?

File: docs/rule-guidelines.md [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+
+For example, specifying the maximum number of digits after the "." in a number
+
+* `number-max-precision`: `int` 
+
+### Space
+
+If the rule is *specifying what space must be* around something then use `*-space` and use `comma`, `colon`, `semicolon`, `opening-brace`, `closing-brace`, `opening-parenthesis`, `closing-parenthesis`, `operator` and `range-operator` to identify that something.
+
+For example, what space must be before a coma and after the operator in a function:
```
Comment (davidtheclark) [inline docs/rule-guidelines.md]: To be consistent with the above: "For example, specifying what ..."

File: docs/rule-guidelines.md [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+
+* "Expected a ... something"
+* "Unexpected ... something" (for rejection e.g. when something is disallowed)
+
+## Tests
+
+Each rule must be accompanied by tests that contain:
+
+* All patterns that are considered warnings.
+* At least one pattern that should *not* be considered a warning.
```
Comment (davidtheclark) [inline docs/rule-guidelines.md]: I'd say that we need to include at least one test-case for all known patterns that should *not* be considered a warning, as well. Changes could break those as easily as the warnings, I'd say.

File: docs/rule-guidelines.md [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+## README.md
+
+Each rule must be accompanied by a README.md, which takes the form of:
+
+1. Rule name.
+2. Single line description.
+3. Code example (if necessary).
+4. Expanded description (if necessary).
+5. Options (if applicable).
+6. Example patterns that are considered warning (for each option value).
```
Comment (davidtheclark) [inline docs/rule-guidelines.md]: "warning" here and next line should be plural.

**Comment by davidtheclark** [comment]
I think it's really good! I threw in a few proofreader-style comments, that's all.

File: docs/rule-guidelines.md (hunk 3 above)
Comment (jeddy3) [inline docs/rule-guidelines.md]: Ha, I always seem to make that typo :)

**Comment by jeddy3** [comment]
Boom. Done.

Thanks for proofing :)

**Comment by MoOx** [comment]
Good works. You rock both of you guys !

**Pull request merged**